### PR TITLE
python310Packages.poetry: 1.2.2 -> 1.3.0

### LINKED ISSUE #204494: python310Packages.poetry is broken on unstable
### Steps To Reproduce
1. build `python310Packages.poetry`

### Build log
```
error: build of '/nix/store/10psvwb5fqcmcdnwfnx2h64jpn5xsbfi-python3.10-poetry-1.2.2.drv' on 'ssh://nix-community-build01' failed: builder for '/nix/store/10psvwb5fqcmcdnwfnx2h64jpn5xsbfi-python3.10-poetry-1.2.2.drv' failed with exit code 1;
       last 10 log lines:
       > Requirement already satisfied: dulwich<0.21.0,>=0.20.46 in /nix/store/1dwv35r9nr6x7hh9s0qfzfijn0ylc44n-python3.10-dulwich-0.20.50/lib/python3.10/site-packages (from poetry==1.2.2) (0.20.50)
       > Requirement already satisfied: html5lib<2.0,>=1.0 in /nix/store/zbi87b6967x988mdriihy6lm195xbib3-python3.10-html5lib-1.1/lib/python3.10/site-packages (from poetry==1.2.2) (1.1)
       > Requirement already satisfied: requests-toolbelt<0.10.0,>=0.9.1 in /nix/store/g6klr9lgfrkxf7wkdsf7r9pc39n45hgd-python3.10-requests-toolbelt-0.9.1/lib/python3.10/site-packages (from poetry==1.2.2) (0.9.1)
       > Requirement already satisfied: crashtest in /nix/store/m1sqz36r1zyjybi9321av7pyafvm9ql7-python3.10-crashtest-0.4.1/lib/python3.10/site-packages (from poetry==1.2.2) (0.4.1)
       > Requirement already satisfied: pexpect<5.0.0,>=4.7.0 in /nix/store/lxinahd9qhdzmqia3493mrgkqh3ingig-python3.10-pexpect-4.8.0/lib/python3.10/site-packages (from poetry==1.2.2) (4.8.0)
       > Requirement already satisfied: poetry-core==1.3.2 in /nix/store/y7m62x2v20cxga1vpqr7kmbgc6nibs5s-python3.10-poetry-core-1.3.2/lib/python3.10/site-packages (from poetry==1.2.2) (1.3.2)
       > Requirement already satisfied: urllib3<2.0.0,>=1.26.0 in /nix/store/fbac0h6j5har9447am8mqswnf1b2mrhh-python3.10-urllib3-1.26.12/lib/python3.10/site-packages (from poetry==1.2.2) (1.26.12)
       > ERROR: Could not find a version that satisfies the requirement cleo<2.0.0,>=1.0.0a5 (from poetry) (from versions: none)
       > ERROR: No matching distribution found for cleo<2.0.0,>=1.0.0a5
       > 
       For full logs, run 'nix log /nix/store/10psvwb5fqcmcdnwfnx2h64jpn5xsbfi-python3.10-poetry-1.2.2.drv'.
error: builder for '/nix/store/10psvwb5fqcmcdnwfnx2h64jpn5xsbfi-python3.10-poetry-1.2.2.drv' failed with exit code 1;
       last 10 log lines:
       > Requirement already satisfied: dulwich<0.21.0,>=0.20.46 in /nix/store/1dwv35r9nr6x7hh9s0qfzfijn0ylc44n-python3.10-dulwich-0.20.50/lib/python3.10/site-packages (from poetry==1.2.2) (0.20.50)
       > Requirement already satisfied: html5lib<2.0,>=1.0 in /nix/store/zbi87b6967x988mdriihy6lm195xbib3-python3.10-html5lib-1.1/lib/python3.10/site-packages (from poetry==1.2.2) (1.1)
       > Requirement already satisfied: requests-toolbelt<0.10.0,>=0.9.1 in /nix/store/g6klr9lgfrkxf7wkdsf7r9pc39n45hgd-python3.10-requests-toolbelt-0.9.1/lib/python3.10/site-packages (from poetry==1.2.2) (0.9.1)
       > Requirement already satisfied: crashtest in /nix/store/m1sqz36r1zyjybi9321av7pyafvm9ql7-python3.10-crashtest-0.4.1/lib/python3.10/site-packages (from poetry==1.2.2) (0.4.1)
       > Requirement already satisfied: pexpect<5.0.0,>=4.7.0 in /nix/store/lxinahd9qhdzmqia3493mrgkqh3ingig-python3.10-pexpect-4.8.0/lib/python3.10/site-packages (from poetry==1.2.2) (4.8.0)
       > Requirement already satisfied: poetry-core==1.3.2 in /nix/store/y7m62x2v20cxga1vpqr7kmbgc6nibs5s-python3.10-poetry-core-1.3.2/lib/python3.10/site-packages (from poetry==1.2.2) (1.3.2)
       > Requirement already satisfied: urllib3<2.0.0,>=1.26.0 in /nix/store/fbac0h6j5har9447am8mqswnf1b2mrhh-python3.10-urllib3-1.26.12/lib/python3.10/site-packages (from poetry==1.2.2) (1.26.12)
       > ERROR: Could not find a version that satisfies the requirement cleo<2.0.0,>=1.0.0a5 (from poetry) (from versions: none)
       > ERROR: No matching distribution found for cleo<2.0.0,>=1.0.0a5
       >
       For full logs, run 'nix log /nix/store/10psvwb5fqcmcdnwfnx2h64jpn5xsbfi-python3.10-poetry-1.2.2.drv'.
```

### Notify maintainers

@jbaum98 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
❯ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.79, NixOS, 23.05 (Stoat), 23.05pre431646.61a8a98e6d5`
 - - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.11.1`
 - channels(root): `"home-manager, nixos, sops-nix"`
 - channels(raito): `"home-manager, nixgl, nixpkgs"`
 - nixpkgs: `/home/raito/.nix-defexpr/channels/nixpkgs`
```

### PR DESCRIPTION
**Pull request by dotlambda**
fixes https://github.com/NixOS/nixpkgs/issues/204494

###### Description of changes
https://github.com/python-poetry/poetry/blob/1.3.0/CHANGELOG.md

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by dotlambda** [comment]
Should this go to staging-next because it fixes `python3Packages.poetry`?

File: pkgs/development/python-modules/poetry/default.nix [hunk 1]
```diff
@@ -82,6 +78,7 @@ buildPythonPackage rec {
     requests-toolbelt
     shellingham
     tomlkit
+    trove-classifiers
```
Comment (tjni) [inline pkgs/development/python-modules/poetry/default.nix:81]: From https://github.com/python-poetry/poetry/commit/b28339d8b76500025fadcd23c87debfc368b609c, I think tomli is needed in Python < 3.11.

**Review by tjni — COMMENTED** [review]
Looks pretty good to me. Thank you for doing this! I guess this does need to go to staging because of the number of rebuilds, though I wonder why so many are needed.

**Comment by dotlambda** [comment]
> though I wonder why so many are needed.

Because this updates poetry-core which is used yo build lots of Python packages.

**Review by tjni — APPROVED** [review]

**Comment by dotlambda** [comment]
Temporarily changed base to master so I can rebase on top of #205402.

**Comment by houstdav000** [comment]
Looks like it might be complicated for poetry2nix; the Poetry lock files now go as far as to specify resource hashes.

Does the `pythonImportsCheckPhase` normally just outright fail the build without an error message? The results in [the most recent x86_64-linux build](https://logs.nix.ci/?key=nixos/nixpkgs.205370&attempt_id=8fe97059-8cab-4b49-b5e6-82b4a8deaed9) seem cryptic to me with the sudden end.

**Comment by dotlambda** [comment]
> Looks like it might be complicated for poetry2nix

This PR is not related to poetry2nix.

**Comment by lockejan** [comment]
Thanks for the work. Could this also be a candidate to be backported to 22.11? 
python3.10-poetry-1.2.2 is known to be affected by CVE-2022-42966 and was marked insecure.

**Comment by dotlambda** [comment]
`poetry` is packaged independently from `python3Packages.poetry`. See https://github.com/NixOS/nixpkgs/pull/205429.

**Pull request merged**